clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -293,12 +293,10 @@ private void closeInternal(final Duration timeout) {
      * Check the unsent queue one last time and poll until all requests are sent or the timer runs out.
      */
     private void sendUnsentRequests(final Timer timer) {
-        if (networkClientDelegate.unsentRequests().isEmpty())
-            return;
-        do {
+        while (timer.notExpired() && networkClientDelegate.hasAnyPendingRequests()) {
```

Comment:
sorry a bit late but one concern: wouldn't this change make that requests that are issue with timeout 0 won't be sent out??? My understanding is that the expected behaviour for requests with timeout 0 is fire-and-forget, so we want to send them at least once.

Reply:
@lianetm you are right. Maybe we can keep origin code but we replace all `networkClientDelegate.unsentRequests().isEmpty()` by `!networkClientDelegate.hasAnyPendingRequests()`?

Reply:
exactly, sounds good to me: keeping the early return `if` to avoid poll, `do-while` to ensure we send out at least once, both conditions using `hasAnyPendingRequests`

Reply:
Hi @lianetm, thank you for pointing out this potential bug, I will fix it. 🙇🏼